bioblend/galaxy/libraries/__init__.py

```diff
@@ -304,6 +330,8 @@ def _do_upload(self, library_id, **keywords):
             payload["roles"] = keywords["roles"]
         if keywords.get("link_data_only", None) and keywords['link_data_only'] != 'copy_files':
             payload["link_data_only"] = 'link_to_files'
+        payload['tag_using_filenames'] = str(keywords.get('tag_using_filenames', True))
```

Comment:
I might be wrong here, but I think this shouldn't be a string ?

Reply:
Not sure... Have you seen this comment from @nsoranzo https://github.com/galaxyproject/bioblend/pull/267#discussion_r214043361 ?

Reply:
@mvdbeek @abretaud Thanks for bringing this up, I've corrected the JSON-encoding for Multipart POST requests in commit https://github.com/galaxyproject/bioblend/commit/c5059e550c54de7fd60a90a48bb627918e42eca7 .